client/client.go

```diff
@@ -442,6 +442,10 @@ func configureAwsClient(ctx context.Context, logger hclog.Logger, awsConfig *Con
 	// Test out retrieving credentials
 	if _, err := awsCfg.Credentials.Retrieve(ctx); err != nil {
 		logger.Error("error retrieving credentials", "err", err)
+		classifiedError := classifyError(err, diag.INTERNAL, nil)
+		if classifiedError[0].Type() == diag.ACCESS {
```

Comment:
@disq is this what you had in mind?

Is there a better way to access the `Type` and create a new diag?

Reply:
Not at all, just edit `classifyError` to match the returned err to a diag of proper kind

Reply:
so:
step 1- determine how to detect these user errors in `classifyError`, using `smithy.APIError.ErrorCode()` or other ways
step 2- create the diag with `diag.WithType(diag.USER)` for these errors inside `classifyError`
step 3- call `classifyError` in all places to turn `error`s into `diag.Diagnostics`

classifyError takes a `fallbackType` argument, so if the error wasn't classified it creates a diag with the given fallback type.

Reply:
Thanks for the detailed explanation @disq. My concern is that we might get this error not during initial setup which could be classified differently. Do we know if it's possible to have the same error as a transient one that we can recover from by retrying?

Reply:
Not sure, possibly (transient error) but we don't retry explicitly, leave it to the aws sdk (which is more than capable)